File: tests/fixtures/_highlight-corpus/sample.php
<?php
// PHP sample
declare(strict_types=1);

namespace Demo;

final class Counter
{
    private int $value = 0;

    public function increment(int $n = 1): int
    {
        $this->value += $n;
        return $this->value;
    }

    public function value(): int
    {
        return $this->value;
    }
}

function main(): int
{
    $contents = file_get_contents('input.txt');
    if ($contents === false) {
        return 1;
    }
    $counter = new Counter();
    foreach (explode("\n", $contents) as $line) {
        if ($line !== '') {
            $counter->increment(strlen($line));
        }
    }
    printf("total bytes: %d\n", $counter->value());
    return 0;
}

exit(main());

File: tests/fixtures/_highlight-corpus/sample.scala
// Scala sample
package demo

import scala.io.Source
import scala.util.Using

final class Counter(private var v: Int = 0):
  def increment(n: Int = 1): Int =
    v += n
    v
  def value: Int = v

@main def run(): Unit =
  val counter = Counter()
  Using.resource(Source.fromFile("input.txt")) { src =>
    src.getLines().filter(_.nonEmpty).foreach { line =>
      counter.increment(line.length)
    }
  }
  println(s"total bytes: ${counter.value}")
